src/Tasks/RoslynCodeTaskFactory/RoslynCodeTaskFactory.cs

```diff
@@ -539,7 +542,14 @@ internal static bool TryResolveAssemblyReferences(TaskLoggingHelper log, RoslynC
                 if (FileSystems.Default.FileExists(reference))
                 {
                     // The path could be relative like ..\Assembly.dll so we need to get the full path
-                    resolvedAssemblyReferences.Add(Path.GetFullPath(reference));
+                    string fullPath = Path.GetFullPath(reference);
+                    string directory = Path.GetDirectoryName(fullPath);
+                    resolvedAssemblyReferences.Add(fullPath);
+#if NETFRAMEWORK
+                    AppDomain.CurrentDomain.AssemblyResolve += (_, eventArgs) => TryLoadAssembly(directory, new AssemblyName(eventArgs.Name).Name);
+#else
+                    AssemblyLoadContext.Default.Resolving += (_, assemblyName) => TryLoadAssembly(directory, assemblyName.Name);
```

Comment:
Can we attach this to the plugin ALC that we're in, instead of `Default`?

Reply:
So I _think_ that would be AssemblyLoadContext.GetLoadContext(typeof(RoslynCodeTaskFactory).Assembly)...

Is that right? I vaguely remember that we started make a separate load context for every task a couple years ago, but I never had a solid understanding of all that was going on.

Reply:
Current version works for dotnet build scenarios; changing to Default to GetLoadContext(typeof(RoslynCodeTaskFactory).Assembly) did not work. Is it something else?

Reply:
What does `CodeTaskFactory` do for .NET Framework? Does it just add it to the default appdomain?

Reply:
Yes:
https://github.com/dotnet/msbuild/blob/9476453a935b612482332633abc4b1c144e1e483/src/Tasks/CodeTaskFactory.cs#L54
(Though it has to flesh out that function.)

Reply:
Team triage:
Try just using AppDomain.CurrentDomain.AssemblyResolve even on Core

Reply:
And remove the new assembly resolving behavior